python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -932,7 +933,9 @@ def get_bin_pack_residual(
 
 def _fits(node: ResourceDict, resources: ResourceDict) -> bool:
     for k, v in resources.items():
-        if v > node.get(k, 0.0):
+        if v > node.get(
+            k, 1.0 if k.startswith(ray_constants.IMPLICIT_RESOURCE_PREFIX) else 0.0
```

Comment:
Q: should we just hide this logic under ResourceDict instead of exposing it here?

Reply:
Same for the below logic

Reply:
Currently ResourceDict is just an alias of python dict. Added a TODO to wrap it in a class.